### PR TITLE
feature/Juan_Sarabino/solid

### PR DESCRIPTION
**Pull request by JSarabino**
Juan Camilo Sarabino Alegria

### CODE REVIEW

File: principles/solid/dependencyInjection.md [hunk 1]
```diff
@@ -1,22 +1,68 @@
...
+        public void start() {
+            engine.start();
+        }
+    }
+
+    public class Engine {
+        public void start() {...}
+    }
+
+The code will work, for now, but what if we wanted to add another engine type, let’s say a diesel engine?
```
Comment (zahydo) [inline principles/solid/dependencyInjection.md]: This is not the right example for Dependency Injection. It's more familiar with Open/Closed

File: principles/solid/liskvsustitution.md [hunk 2]
```diff
@@ -1,22 +1,75 @@
...
 
-[TODO]
+The Liskov Substitution Principle (LSP) applies to inheritance hierarchies 
+such that derived classes must be completely substitutable for their base classes.
+Consider a typical example of a Square derived class and Rectangle base class:
 
 ### Bad example:
 
-[TODO]
+    public class Rectangle {
```
Comment (zahydo) [inline principles/solid/liskvsustitution.md]: Wrong code format. Use ```java to format in Markdown

**Review by zahydo — CHANGES_REQUESTED** [review]
The descriptions of the principles must include the problem, how it solves, and the benefits. Also, the examples are very simple.

**Comment by zahydo** [comment]
Change the name of the Pull Request with the name of the branch

File: principles/solid/liskvsustitution.md (hunk 2 above)
Comment (JSarabino) [reply principles/solid/liskvsustitution.md]: i didn't knew the Markdown format, thanks!

**Comment by zahydo** [comment]
Checked

**Pull request closed without merging**